src/Api/Domain.php

```diff
@@ -88,16 +88,26 @@ public function create(string $domain, string $smtpPass = null, string $spamActi
 
         $params['name'] = $domain;
 
-        // If at least smtpPass available, check for the fields spamAction wildcard
         if (!empty($smtpPass)) {
+            Assert::stringNotEmpty($smtpPass);
+            Assert::lengthBetween($smtpPass, 5, 32, 'SMTP password must be between 5 and 32 characters.');
+
+            $params['smtp_password'] = $smtpPass;
+        }
+
+        if (null !== $spamAction) {
```

Comment:
I would suggest keeping here the `!empty` approach due to:
- `''` will pass this check
- to be consistent with other checks
```suggestion
        if (!empty($spamAction)) {
```